src/Collection.php

```diff
@@ -624,7 +624,7 @@ public function isEmpty(): bool
      */
     public function jsonSerialize(): array
     {
-        return $this->all();
+        return $this->all(false);
```

Comment:
we cannot normalize by default in `jsonSerialize` because that would break usage with a "map" collection, e.g.:

```PHP
// Correct -> using `all(false)`
json_encode(Collection::fromIterable(['foo' => 1, 'bar' => 2])); // '{"foo": 1, "bar": 2}'

// Incorrect -> using the new `all()`
json_encode(Collection::fromIterable(['foo' => 1, 'bar' => 2])); // '[1, 2]'
```

@drupol also I realised there's no mention in the documentation that Collection can be serialized. I am thinking to write about it and mention this behaviour as well. What do you think is the best place for it? I would say we could mention it in 2 places:
- `usage.rst` -> have a short "Serialization" section
- `api.rst` -> mentioning the `jsonSerialize` method

Reply:
Oh! Nice catch.

I would do it in both files - as serialization is a feature AND part of the API.